docs/en/guides/pipeline-integrations/administration/config-staging-and-rollout.md

```diff
@@ -6,3 +6,489 @@ lang: en
+_Please note that this document describes functionality only available if you have taken control over a Toolkit configuration. For the default setup, please see  [{% include product %} Integrations Admin Guide](https://support.shotgunsoftware.com/hc/en-us/articles/115000067493)._
+
+# Introduction
+
+This document outlines how to manage your Toolkit configuration. Toolkit contains several utilities and tools that lets you safely manage your configuration and test upgrades and changes in a safe and sandboxed environment before rolling them out to the entire production.
+
+In this document, we'll describe how to:
+
+-   Safely upgrade the Toolkit Core API.
+-   Upgrading your Apps and Engines.
+-   Various ways to manage your config across multiple projects.
+-   Go through Toolkit's  _Clone_  and  _Push_  functionality that allows you to safely test upgrades and changes without disrupting production.
+-   How to work with  `git`  source control and Toolkit.
+
+# Basics of configuration management
+
+Each Toolkit Project has one or more configurations associated with it. The configuration contains all the settings that the project needs: file system locations (templates), engines apps etc. In toolkit, each project is independently configurable. This means that different projects can live side by side safely: upgrading one project will not affect another.
+
+If you are working with a large number of projects, this may become cumbersome and we offer several ways to make this process easy, safe and streamlined.
+
+In {% include product %}, each project has a number of  **Pipeline Configurations**. When a project is first set up with Toolkit, a Pipeline Configuration called  `primary`  is created. The pipeline configuration entity in {% include product %} points at a location on disk where the toolkit configuration can be found.
+
+During the course of a project, you often need to make changes to the configuration. This can be tweaks to the configuration, or perhaps you need to add additional apps or engines. We also release new app versions frequently and we recommend that you use the latest versions if possible.
+
+While it is possible to upgrade your  _primary_  project configuration straight away, this can be risky: Since this configuration is used by everybody on the project, introducing a problem will affect everyone. A better approach is to create an isolated version of the configuration that a select group of people have access to. In this safe environment, upgrades, configuration changes and development can happen without affecting the rest of the production. Once the changes have been tested, they can be safely and confidently pushed to the primary configuration.
```

Comment:
```suggestion
While it is possible to upgrade your  _primary_  project configuration straight away, this can be risky; since this configuration is used by everybody on the project, introducing a problem will affect everyone. A better approach is to create an isolated version of the configuration that a select group of people have access to. In this safe environment, upgrades, configuration changes and development can happen without impacting the rest of the production. Once the changes have been tested, they can be safely and confidently pushed to the primary configuration.
```